main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -682,7 +682,7 @@ object Resolver extends Phase[NamedAst.Root, ResolvedAst.Root] {
                   // If the tag is `Some` we construct the lambda: x -> Some(x).
 
                   // Construct a fresh symbol for the formal parameter.
-                  val freshVar = Symbol.freshVarSym("x", BoundBy.FormalParam, loc)
+                  val freshVar = Symbol.freshVarSym(Flix.Delimiter + "x", BoundBy.FormalParam, loc)
```

Comment:
I am worried about names starting with the delimiter. I would go for `tmp$delim` or whatever.

Reply:
How about ending with the delimiter? It looks like that's what we do in the Weeder.

Reply:
Thats fine